### ISSUE TITLE
TODO - validateStatus fn to determine which path[xxx].responses[yyy] should be the main success

### ISSUE THREAD

**Issue by astahmer** [astahmer/openapi-zod-client#12]
## the problem: 
after trying lots of OpenAPI specs, I noticed that sometimes there would be no `ResponseObject` defined with "200" or "default" (in the `responses` map of an `OperationObject`), but rather return a 201, 202 (or basically anything else than 200/default)

when this happens, `openapi-zod-client` would therefore not output anything for that endpoint

## example / reproduction

using the [petstore example](https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/petstore.yaml) as base, here is a minimal reproduction

input:
```yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /pets:
    post:
      summary: Create a pet
      operationId: createPets
      tags:
        - pets
      responses:
        "201":
          description: Expected response to a valid request
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Pet"
components:
  schemas:
    Pet:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
        tag:
          type: string

```

output:

```ts
import { Zodios } from "@zodios/core";
import { z } from "zod";

const v7LgRCMpuZ0 = z.object({ id: z.bigint(), name: z.string(), tag: z.string().optional() }).optional();

const variables = {
    createPets: v7LgRCMpuZ0,
};

const endpoints = [] as const;

export const api = new Zodios(endpoints);
```

whereas if we just replace "201" with "200", the output becomes:

```ts
import { Zodios } from "@zodios/core";
import { z } from "zod";

const v7LgRCMpuZ0 = z.object({ id: z.bigint(), name: z.string(), tag: z.string().optional() }).optional();

const variables = {
    Pet: v7LgRCMpuZ0,
    createPets: v7LgRCMpuZ0,
};

const endpoints = [
    {
        method: "post",
        path: "/pets",
        requestFormat: "json",
        response: variables["Pet"],
    },
] as const;

export const api = new Zodios(endpoints);
```

## solution

basically do it [just like axios](https://github.com/axios/axios#handling-errors)

default behavior: if "200" or "default" are not listed in the `ResponsesObject` map, then try each status `status >= 200 && status < 300` ? 
controlled behaviour: let the user determine that by passing a `validateStatus` (still just like axios) if needed in the `TemplateContext`.options
